db/migrate/20160815160730_create_functions.rb

```diff
@@ -0,0 +1,12 @@
+class CreateFunctions < ActiveRecord::Migration[5.0]
+  def change
+    create_table :functions do |t|
+      t.belongs_to :user, foreign_key: true
+      t.belongs_to :role, foreign_key: true
+      t.date :assumed_at
+      t.date :vacated_at
```

Comment:
I recently read in a styleguide:
* use suffix `_at` for `datetime`
* use suffix `_on` for `date`
* use suffix `_time` for `time`
Which makes a lot of sense to me :-)